### PR TITLE
feat(batch): add secrets props to job definition

### LINKED ISSUE #10976: [Batch] Add "Secrets" properties in interface JobDefinitionContainer
### Use Case

Need to pass Secrets to a Batch Job with CDK Batch Construct.
Not available yet : 
https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-batch.JobDefinitionContainer.html

### Proposed Solution

Like with CloudFormation, it'll be interesting to add "Secrets" propertie to the CDK Batch Construct : 
https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-batch.CfnJobDefinition.ContainerPropertiesProperty.html#secrets

### Other

https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-batch.CfnJobDefinition.SecretProperty.html

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by yoshizawa56**
Add a secrets property to batch.JobDefinitionContainer. This interface is almost the same as ecs.ContainerDefinitionOptions.

This is reopen PR of #19506

closes #10976

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by kaizencc — COMMENTED** [review]
Hi @yoshizawa56! Is this the same PR that was previously abandoned (by us, sorry)? If so, can you link to that in the PR description as well as provide a summary of what's in this PR, also in the description. I'll make sure this one gets reviewed promptly.

**Comment by yoshizawa56** [comment]
Hi @kaizencc ! Thanks for commnet!
This is the same PR as previous one(https://github.com/aws/aws-cdk/pull/19506). I edited the summary.

File: packages/@aws-cdk/aws-batch/test/integ.batch.expected.json [hunk 1]
```diff
@@ -0,0 +1,1874 @@
+{
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-batch/test/integ.batch.expected.json:1]: I'm 90% sure that this file is not needed and shouldn't be a part of this PR. `expected.json` files are a relic of our legacy integration test system which are now replaced by the `batch.integ.snapshot` folder. I am not sure why/how this file was created, but I suspect it has something to do with the initial PR (from March) coming before we introduced the new integ system.

Can you please delete this file and then run `yarn integ` in your batch folder again? This file should not get created and that will verify what I'm saying is true.

**Review by kaizencc — CHANGES_REQUESTED** [review]
@yoshizawa56 one minor thing. Otherwise this looks great!

File: packages/@aws-cdk/aws-batch/test/integ.batch.expected.json (hunk 1 above)
Comment (yoshizawa56) [reply packages/@aws-cdk/aws-batch/test/integ.batch.expected.json]: You are correct. I delete expected.json and re-ran `yarn integ` and it passed the test!

File: packages/@aws-cdk/aws-batch/README.md [hunk 2]
```diff
@@ -300,6 +300,23 @@ new batch.JobDefinition(this, 'job-def', {
...
+
+```ts
+const dbSecret = new secretsmanager.Secret(this, 'secret');
+
+new batch.JobDefinition(this, 'batch-job-def-secrets', {
+  container: {
+    image: ecs.EcrImage.fromRegistry('docker/whalesay'),
+    secrets: {
+      PASSWORD: ecs.Secret.fromSecretsManager(dbSecret, 'password'),
+    }
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-batch/README.md:315]:
```suggestion
    },
```

**Review by kaizencc — APPROVED** [review]
Awesome work @yoshizawa56! Thanks for the PR.

**Review by kaizencc — APPROVED** [review]

**Pull request merged**